app/controllers/topics_controller.rb

```diff
@@ -397,7 +397,13 @@ def update
       bypass_bump = should_bypass_bump?(changes)
 
       first_post = topic.ordered_posts.first
-      success = PostRevisor.new(first_post, topic).revise!(current_user, changes, validate_post: false, bypass_bump: bypass_bump)
+      success = PostRevisor.new(first_post, topic).revise!(
+        current_user,
+        changes,
+        validate_post: false,
+        bypass_bump: bypass_bump,
+        keep_existing_draft: [true, "true"].include?(params[:keep_existing_draft])
```

Comment:
```suggestion
        keep_existing_draft: params[:keep_existing_draft].to_s == "true"
```
This has less cases to deal with and seems simpler to me.

Reply:
That's better indeed, thanks!